src/lib/utils/arbProvider.ts

```diff
@@ -54,46 +55,38 @@ class ArbFormatter extends Formatter {
 }
 
 /**
- * Fetch a transaction receipt from an l2Provider
- * Additional batch info is also returned if requested
- * @param l2Provider
- * @param txHash
- * @returns
+ * Arbitrum specific formats
  */
-export async function getArbTransactionReceipt(
-  l2Provider: JsonRpcProvider,
-  txHash: string
-): Promise<ArbTransactionReceipt | null> {
-  const rec = await l2Provider.send('eth_getTransactionReceipt', [txHash])
-  if (rec == null) return null
-  const arbFormatter = new ArbFormatter()
-  return arbFormatter.receipt(rec)
-}
+export class ArbitrumProvider extends JsonRpcProvider {
+  private static arbFormatter = new ArbFormatter()
 
-/**
- * Fetch a block for the provided hash, including additional arbitrum specific fields
- * @param l2Provider
- * @param blockHash
- * @param includeTransactions
- */
-export async function getArbBlockByHash<T extends boolean = false>(
-  l2Provider: JsonRpcProvider,
-  blockHash: string,
-  includeTransactions?: T
-): Promise<T extends true ? ArbBlockWithTransactions | null : ArbBlock | null>
-export async function getArbBlockByHash<T extends boolean = false>(
-  l2Provider: JsonRpcProvider,
-  blockHash: string,
-  includeTransactions?: T
-): Promise<ArbBlock | ArbBlockWithTransactions | null> {
-  const l2Block = await l2Provider.send('eth_getBlockByHash', [
-    blockHash,
-    includeTransactions,
-  ])
-  if (l2Block === null) return null
-  const arbFormatter = new ArbFormatter()
+  public constructor(provider: JsonRpcProvider, network?: Networkish) {
+    super(provider.connection, network)
+  }
+
+  static override getFormatter(): Formatter {
+    return this.arbFormatter
+  }
 
-  return includeTransactions
-    ? arbFormatter.blockWithTransactions(l2Block)
-    : (arbFormatter.block(l2Block) as unknown as ArbBlockWithTransactions)
+  public override async getTransactionReceipt(
+    transactionHash: string | Promise<string>
+  ): Promise<ArbTransactionReceipt> {
+    return (await super.getTransactionReceipt(
+      transactionHash
+    )) as ArbTransactionReceipt
```

Comment:
we could sanity check for the inclusion of the arb-specific fields before the type assertions here (and the other 2 functions below) (and maybe throw an error if not included? this would just be to prevent confusion if an ArbProvider isn't really an Arbitrum node, maybe unnecessary)